api/services/SiteCreator.js

```diff
@@ -82,6 +82,7 @@ const createSiteFromExistingRepo = ({ siteParams, user }) => {
 const createSiteFromTemplate = ({ siteParams, user, template }) => {
   let site = Site.build(siteParams)
   site.engine = "jekyll"
+  site.defaultBranch = templateForTemplateName(template).branch
```

Comment:
We provide the template, the client just provides the template name. If the client attempts to provide a template name for a template that does not exist, we error before we get here: https://github.com/18F/federalist/pull/800/files#diff-6eb3478dc1e9d3885c6e6431de4931f5R134